lib/fetch/formdata.js

```diff
@@ -0,0 +1,176 @@
+  }
+
+  set (name, value, filename) {
+    // The set(name, value) and set(name, blobValue, filename) method steps
+    // are:
+
+    // 1. Let value be value if given; otherwise blobValue.
+    // 2. Let entry be the result of creating an entry with name, value, and
+    // filename if given.
+    const entry = makeEntry(name, value, filename)
+
+    // 3. If there are entries in this’s entry list whose name is name, then
+    // replace the first such entry with entry and remove the others.
+    this[kState] = this[kState].filter(entry => entry.name !== name)
+
+    // 4. Otherwise, append entry to this’s entry list.
+    this[kState].push(entry)
+  }
+
+  * [Symbol.iterator] () {
+    // The value pairs to iterate over are this’s entry list’s entries with
+    // the key being the name and the value being the value.
+    for (const { name, value } of this[kState]) {
+      yield [name, value]
+    }
+  }
+}
+
+function makeEntry (name, value, filename) {
+  // To create an entry for name, value, and optionally a filename, run these
+  // steps:
+
+  // Let entry be a new entry.
+  const entry = {
+    name: null,
+    value: null
+  }
+
+  // Set entry’s name to name.
+  entry.name = name
+
+  // If value is a Blob object and not a File object, then set value to a new File
+  // object, representing the same bytes, whose name attribute value is "blob".
+  if (value instanceof Blob && !(value instanceof File)) {
+    value = new File(value, 'blob')
+  }
+
+  // If value is (now) a File object and filename is given, then set value to a
+  // new File object, representing the same bytes, whose name attribute value is
+  // filename.
+  if (value instanceof File && filename) {
+    value = new File(value, filename)
+  }
+
+  // Set entry’s value to value.
+  entry.value = value
+
+  // Return entry.
+  return entry
+}
+
+/*! formdata-polyfill. MIT License. Jimmy Wärting <https://jimmy.warting.se/opensource> */
+const escape = (str, filename) =>
+  (filename ? str : str.replace(/\r?\n|\r/g, '\r\n'))
+    .replace(/\n/g, '%0A')
+    .replace(/\r/g, '%0D')
+    .replace(/"/g, '%22')
+
+function formDataToBlob (formData) {
+  const boundary = ('----formdata-polyfill-' + Math.random())
+  const chunks = []
+  const prefix = `--${boundary}\r\nContent-Disposition: form-data; name="`
+
+  for (const [name, value] of formData) {
+    if (typeof value === 'string') {
+      chunks.push(prefix + escape(name) + `"\r\n\r\n${value.replace(/\r(?!\n)|(?<!\r)\n/g, '\r\n')}\r\n`)
+    } else {
+      chunks.push(
+        prefix + escape(name) + `"; filename="${escape(value.name, 1)}"\r\n` +
+        `Content-Type: ${value.type || 'application/octet-stream'}\r\n\r\n`,
+        value,
+        '\r\n'
+      )
+    }
+  }
+
+  chunks.push(`--${boundary}--`)
+
+  return new Blob(chunks, {
+    type: 'multipart/form-data; boundary=' + boundary
+  })
+}
+
+function blobToFormData (blob) {
+  // TODO
+  return new FormData()
```

Comment:
Still missing parsing :(

Reply:
Where is the spec for this?

Reply:
https://xhr.spec.whatwg.org/#formdata

Reply:
https://datatracker.ietf.org/doc/html/rfc7578